code/modules/client/preference/preferences.dm

```diff
@@ -1278,11 +1283,14 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 
 					species = input("Please select a species", "Character Generation", null) in new_species
 					var/datum/species/NS = all_species[species]
+					if(!istype(NS)) //The species was invalid. Set the species to the default and fetch the datum for that species.
+						species = initial(species)
```

Comment:
In this case, I believe it would be better to reject the user's input, and inform them that whatever they chose is not a species.